### PR TITLE
VC's did/will (dis)appear

### PR DESCRIPTION
**Pull request by RuiAAPeres**

### CODE REVIEW

File: Source/UIKit/UIViewController.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    /// Returns a `Signal`, that will be triggered
+    /// when `self`'s `viewWillDisappear` is called
+    public var rex_viewWillDisappearSignal: Signal<(), NoError> {
+        return triggerForSelector(#selector(UIViewController.viewWillDisappear(_:)))
+    }
+    
+    private func triggerForSelector(selector: Selector) -> Signal<(), NoError>  {
+        return self
+            .rac_signalForSelector(selector)
+            .rex_toSignal()
```
Comment (iv-mexx) [inline Source/UIKit/UIViewController.swift]: Use `rex_toTriggerSignal()` from #99 ?

**Comment by mdiep** [comment]
It might be nice to add the _appear_ methods while you're at it. It seems natural to want both if you want one.

File: Source/UIKit/UIViewController.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        return triggerForSelector(#selector(UIViewController.viewDidAppear(_:)))
+    }
+    
+    /// Returns a `Signal`, that will be triggered
+    /// when `self`'s `viewWillAppear` is called
+    public var rex_viewWillAppearSignal: Signal<(), NoError> {
+        return triggerForSelector(#selector(UIViewController.viewWillAppear(_:)))
+    }
+    
+    private func triggerForSelector(selector: Selector) -> Signal<(), NoError>  {
```
Comment (335g) [inline Source/UIKit/UIViewController.swift:38]: Should `triggerForSelector` be inside extension of `NSObject` (or `UIResponder`) ?

Comment (RuiAAPeres) [inline Source/UIKit/UIViewController.swift:38]: @335g Looking at `UIResponder` documentation, I don't think so. I might be missing something.

Comment (335g) [inline Source/UIKit/UIViewController.swift:38]: Now, It might be good in this 👍

**Comment by neilpa** [comment]
Awesome!

**Pull request merged**